src/smoothl1loss/problem_description.cpp

```diff
@@ -0,0 +1,165 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <miopen/smoothl1loss/problem_description.hpp>
+#include <miopen/names.hpp>
+
+#include <sstream>
+
+namespace miopen {
+
+namespace smoothl1loss {
+
+bool checkSameType(const TensorDescriptor& x, const TensorDescriptor& y)
+{
+    if(x.GetType() != y.GetType())
+    {
+#if MIOPEN_BUILD_DEV || !MIOPEN_NDEBUG
+        MIOPEN_THROW(miopenStatusBadParm, "Reduce: Tensor types do not match.");
+#else
+        return false;
+#endif
+    }
+    return true;
+}
+
+bool checkSameLength(const TensorDescriptor& x, const TensorDescriptor& y)
+{
+    if(x.GetSize() != y.GetSize())
+#if MIOPEN_BUILD_DEV || !MIOPEN_NDEBUG
+        MIOPEN_THROW(miopenStatusBadParm, "Smooth L1Loss: Tensor sizes do not match.");
+#else
+        return false;
+#endif
+    for(int32_t i = 0; i < x.GetSize(); ++i)
+    {
+        if(x.GetLengths()[i] != y.GetLengths()[i])
+        {
+#if MIOPEN_BUILD_DEV || !MIOPEN_NDEBUG
+            MIOPEN_THROW(miopenStatusBadParm, "Smooth L1Loss: Tensor sizes do not match.");
+#else
+            return false;
+#endif
+        }
+    }
+    return true;
+}
+
+bool checkSameStride(const TensorDescriptor& x, const TensorDescriptor& y)
+{
+    if(x.GetSize() != y.GetSize())
+#if MIOPEN_BUILD_DEV || !MIOPEN_NDEBUG
+        MIOPEN_THROW(miopenStatusBadParm, "Smooth L1Loss: Tensor strides do not match.");
+#else
+        return false;
+#endif
+    for(int32_t i = 0; i < x.GetSize(); ++i)
+        if(x.GetStrides()[i] != y.GetStrides()[i])
+#if MIOPEN_BUILD_DEV || !MIOPEN_NDEBUG
+            MIOPEN_THROW(miopenStatusBadParm, "Smooth L1Loss: Tensor strides do not match.");
+#else
+            return false;
+#endif
+    return true;
+}
+
+bool checkRightStride(const TensorDescriptor& x)
+{
+    auto strides = x.GetStrides();
+    auto lengths = x.GetLengths();
+    std::vector<std::pair<size_t, size_t>> p;
+    p.reserve(x.GetSize());
+    std::transform(strides.begin(),
+                   strides.end(),
+                   lengths.begin(),
+                   std::back_inserter(p),
+                   [](size_t a, size_t b) { return std::make_pair(a, b); });
+    std::sort(p.begin(), p.end());
+    for(int i = 1; i < p.size(); ++i)
+    {
+        if(p[i].first != p[i - 1].first * p[i - 1].second)
+#if MIOPEN_BUILD_DEV || !MIOPEN_NDEBUG
+            MIOPEN_THROW(miopenStatusBadParm, "Smooth L1Loss: Tensor strides do not match.");
+#else
+            return false;
+#endif
+    }
+    return true;
+}
+
+bool checkContiguous(const TensorDescriptor& x)
+{
+    size_t s = 1;
+    for(int i = x.GetSize() - 1; i >= 0; --i)
+    {
+        if(s != x.GetStrides()[i])
+        {
+#if MIOPEN_BUILD_DEV || !MIOPEN_NDEBUG
+            MIOPEN_THROW(miopenStatusBadParm, "Smooth L1Loss: Non-contiguous Tensor.");
```

Comment:
The test failure seems to be because of this. 
It passes in release build but fails in debug build.

Reply:
Oh, I see. I'll fix it

Reply:
Done. Please take a look.